src/java/com/zimbra/qa/selenium/projects/ajax/tests/calendar/trash/GetAppointment.java

```diff
@@ -40,24 +40,35 @@ public GetAppointment() {
 
 		// All tests start at the Calendar page
 		super.startingPage = app.zPageCalendar;
+		super.startingAccountPreferences.put("zimbraPrefCalendarInitialView", "week");
 	}
 
 	@Test( description = "Verify the presence of appointment in Trash after deletion.",
 			groups = { "smoke", "L0" })
 	
 	public void GetAppointment_01() throws HarnessException {
 
-		// Create the appointment on the server
 		// Create the message data to be sent
 		String apptSubject = ConfigProperties.getUniqueString();
 		String location = "location" + ConfigProperties.getUniqueString();
 		String content = "content" + ConfigProperties.getUniqueString();
 
 		// Absolute dates in UTC zone
 		Calendar now = Calendar.getInstance();
-		ZDate startUTC = new ZDate(now.get(Calendar.YEAR), now.get(Calendar.MONTH) + 1, now.get(Calendar.DAY_OF_MONTH), 12, 0, 0);
-		ZDate endUTC   = new ZDate(now.get(Calendar.YEAR), now.get(Calendar.MONTH) + 1, now.get(Calendar.DAY_OF_MONTH), 14, 0, 0);
-
+		ZDate startUTC = null;
+		ZDate endUTC  = null;
+		
+		// if current day of the week is Sunday, create an appointment on Monday so that it remains in the current week view even after time zone adjustment.
+		if ( now.get(Calendar.DAY_OF_WEEK) == 1 ) {
+			
+			startUTC = new ZDate(now.get(Calendar.YEAR), now.get(Calendar.MONTH) + 1, now.get(Calendar.DAY_OF_MONTH) + 1, 12, 0, 0);
```

Comment:
can we simplify the if-else block. Just storing the day of month in a variable based on the current day and then initialising the startUTC & endUTC?